releasenotes/notes/add-clifford-t-transpiler-pipeline-e640b5ba64eab13a.yaml

```diff
@@ -0,0 +1,41 @@
+---
+features_transpiler:
+  - |
+    The function :func:`.generate_preset_pass_manager` now generates a special
+    pass manager when the basis set consists of Clifford+T gates only.
+    Formally, a Clifford+T basis set must contain either a :class:`.TGate` or
+    :class:`.TdgGate` (or both), and only Clifford gates in addition.
+    The full list of supported Clifford gates can be obtained using
+    :func:`.get_clifford_gate_names`.
+
+    For example::
+  
+      from qiskit.circuit import QuantumCircuit
+      from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+      from qiskit.quantum_info import get_clifford_gate_names
+
+      basis_gates = get_clifford_gate_names() + ["t", "tdg"]
```

Comment:
perhaps in the example we would like to fix the basis gates, e.g. `[cx, h, t, tdg]`

Reply:
btw, do we assume that the clifford gates are hard-coded? can the user define their own clifford gate and provide it (say as clifford tableaux)?

Reply:
One of the points of the example is to use `get_clifford_gate_names` to get the list of all standard Clifford gate names  in Qiskit. Most users will probably want to include as many Clifford gates as possible, as this example illustrates this. Of course, the users can specify their own gate lists, such as ``["h", "t", "tdg"]``or even ``["h", "t"]``, please see the last paragraph in the release notes.

So far in all of the transpiler passes ``basis_gates`` is a list of strings, and we probably do not want to generalize this. However, your comment is very relevant for extending the `Target` class for supporting Clifford+T gates in a form that is hopefully better than just listing all the Clifford gate names.